carmin-server/server/test/resources/test_execution.py

```diff
@@ -39,3 +43,94 @@ def test_get_invalid_execution(self, test_config):
+            })
+        execution = load_json_data(response)
+        assert execution["name"] == PATCH_VALID_EXECUTION["name"]
+        assert execution["timeout"] == PATCH_VALID_EXECUTION["timeout"]
+
+    def test_put_valid_execution_name(self, test_config, execution_id):
+        response = test_config.test_client.put(
+            '/executions/{}'.format(execution_id),
+            headers={"apiKey": standard_user().api_key},
+            data=json.dumps(PATCH_VALID_EXECUTION2))
+        assert response.status_code == 204
+
+        response = test_config.test_client.get(
+            '/executions/{}'.format(execution_id),
+            headers={
+                "apiKey": standard_user().api_key
+            })
+        execution = load_json_data(response)
+        assert execution["name"] == PATCH_VALID_EXECUTION2["name"]
+
+    def test_put_valid_execution_timeout(self, test_config, execution_id):
+        response = test_config.test_client.put(
+            '/executions/{}'.format(execution_id),
+            headers={"apiKey": standard_user().api_key},
+            data=json.dumps(PATCH_VALID_EXECUTION3))
+        assert response.status_code == 204
+
+        response = test_config.test_client.get(
+            '/executions/{}'.format(execution_id),
+            headers={
+                "apiKey": standard_user().api_key
+            })
+        execution = load_json_data(response)
+        assert execution["timeout"] == PATCH_VALID_EXECUTION3["timeout"]
+
+    def test_put_illegal_parameter_status(self, test_config, execution_id):
+        response = test_config.test_client.put(
+            '/executions/{}'.format(execution_id),
+            headers={"apiKey": standard_user().api_key},
+            data=json.dumps(PATCH_ILLEGAL_PARAMETER))
+        error = error_from_response(response)
+        expected_error_code_and_message = copy.deepcopy(error)
+        expected_error_code_and_message.error_message = error.error_message.format(
+            "status")
+        assert error == expected_error_code_and_message
+
+    def test_put_illegal_parameter_identifier(self, test_config, execution_id):
+        response = test_config.test_client.put(
+            '/executions/{}'.format(execution_id),
+            headers={"apiKey": standard_user().api_key},
+            data=json.dumps(PATCH_ILLEGAL_PARAMETER2))
+        error = error_from_response(response)
+        expected_error_code_and_message = copy.deepcopy(error)
+        expected_error_code_and_message.error_message = error.error_message.format(
+            "identifier")
+        assert error == expected_error_code_and_message
+
+    def test_put_invalid_parameter(self, test_config, execution_id):
+        response = test_config.test_client.put(
+            '/executions/{}'.format(execution_id),
+            headers={"apiKey": standard_user().api_key},
+            data=json.dumps(PATCH_INVALID_PARAMETER))
+        assert response.status_code == 400
```

Comment:
This 400 is caused by an exception in the DB right? I that case I believe you should add a try catch as done with register, and return a proper ErrorCodeAndMessage.

Reply:
No, the 400 is caused by the unmarshaling of the request. The model is invalid, since timeout should be an integer.